encoder-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -0,0 +1,212 @@
+        .setValue(AnyValue.newBuilder().setIntValue(value))
+        .build();
+  }
+
+  private Span.Builder builderForSingleSpan(MutableSpan span, Builder resourceSpansBuilder) {
+    Span.Builder spanBuilder = Span.newBuilder()
+        .setTraceId(span.traceId() != null ? ByteString.fromHex(span.traceId()) : INVALID_TRACE_ID)
+        .setSpanId(span.id() != null ? ByteString.fromHex(span.id()) : INVALID_SPAN_ID);
+    if (span.name() != null) {
+      spanBuilder.setName(span.name());
+    }
+    if (span.parentId() != null) {
+      spanBuilder.setParentSpanId(ByteString.fromHex(span.parentId()));
+    }
+    long start = span.startTimestamp();
+    long finish = span.finishTimestamp();
+    spanBuilder.setStartTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(start));
+    spanBuilder.setEndTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(finish));
+    Kind kind = span.kind();
+    if (kind != null) {
+      switch (kind) {
+        case CLIENT:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CLIENT);
+          break;
+        case SERVER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_SERVER);
+          break;
+        case PRODUCER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_PRODUCER);
+          break;
+        case CONSUMER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CONSUMER);
+          break;
+      }
+    }
+    else {
+      spanBuilder.setKind(DEFAULT_KIND);
+    }
+    String localServiceName = span.localServiceName();
+    if (localServiceName == null) {
+      localServiceName = Resource.getDefault().getAttribute(ServiceAttributes.SERVICE_NAME);
+    }
+    resourceSpansBuilder.getResourceBuilder().addAttributes(stringAttribute(ServiceAttributes.SERVICE_NAME.getKey(), localServiceName));
+    maybeAddStringAttribute(spanBuilder, NetworkAttributes.NETWORK_LOCAL_ADDRESS.getKey(), span.localIp());
+    maybeAddIntAttribute(spanBuilder, NetworkAttributes.NETWORK_LOCAL_PORT.getKey(), span.localPort());
+    maybeAddStringAttribute(spanBuilder, NetworkAttributes.NETWORK_PEER_ADDRESS.getKey(), span.remoteIp());
+    maybeAddIntAttribute(spanBuilder, NetworkAttributes.NETWORK_PEER_PORT.getKey(), span.remotePort());
+    maybeAddStringAttribute(spanBuilder, PEER_SERVICE, span.remoteServiceName());
+    span.forEachTag(consumer, spanBuilder);
+    span.forEachAnnotation(consumer, spanBuilder);
+    consumer.addErrorTag(spanBuilder, span);
+
+    return spanBuilder;
+  }
+
+  private static void maybeAddStringAttribute(Span.Builder spanBuilder, String key, String value) {
+    if (value != null) {
+      spanBuilder.addAttributes(stringAttribute(key, value));
+    }
+  }
+
+  private static void maybeAddIntAttribute(Span.Builder spanBuilder, String key, int value) {
+    if (value != 0) {
+      spanBuilder.addAttributes(intAttribute(key, value));
+    }
+  }
+
+  static final class Consumer implements TagConsumer<Span.Builder>, AnnotationConsumer<Span.Builder> {
+
+    private final Tag<Throwable> errorTag;
```

Comment:
nit as unimportant, but some cases there are more bytecode for private symbols.

```suggestion
    final Tag<Throwable> errorTag;
```

Reply:
Fixed in https://github.com/openzipkin-contrib/zipkin-otel/pull/9/commits/89cedb02fc0b39c0915f683bbee733afffde55c5